components/calendar/src/cal/hebrew.rs

```diff
@@ -322,21 +322,18 @@ impl Calendar for Hebrew {
     fn month(&self, date: &Self::DateInner) -> MonthInfo {
         let ordinal = date.0.month();
         let standard = self.month_from_ordinal(date.0.year(), ordinal);
-        let formatting = if standard.number() == 6 && ordinal == 7 {
-            const { Month::leap(6) }
-        } else {
-            standard
-        };
+        let standard_code = standard.code();
 
         #[allow(deprecated)] // field-level allows don't work at 1.83 MSRV
         types::MonthInfo {
             ordinal,
-            standard,
-            formatting,
+            value: standard,
+            // Use the leap name for Adar in a leap year
+            format_as_leap: standard.number() == 6 && ordinal == 7,
             #[allow(deprecated)]
-            standard_code: standard.code(),
+            standard_code,
             #[allow(deprecated)]
-            formatting_code: formatting.code(),
+            formatting_code: standard_code,
```

Comment:
Issue: Since this is public API, let's keep the behavior stable.

`format_as_leap` is a new way for `icu_datetime` to get this info, but other crates should be able to get it, too. I want `icu_datetime` to be using the public interface or the equivalent of the public interface. If they can get the equivalent information from `formatting_code`, it's fine.